src/server/pfs/pfs.go

```diff
@@ -281,10 +298,18 @@ func (e ErrSquashWithoutChildren) Error() string {
 	return fmt.Sprintf("cannot squash a commit that has no children as that would cause data loss, use the drop operation instead: %s", e.Commit)
 }
 
+func (e ErrDeleteWithDependentCommitSets) Error() string {
+	return fmt.Sprintf("the commit sets %v, cannot be squashed in isolation. To delete them, also squash: %v", e.RequestedDeleteCommitSets, e.MinimalCommitSets)
+}
+
 func (e ErrDropWithChildren) Error() string {
 	return fmt.Sprintf("cannot drop a commit that has children: %s", e.Commit)
 }
 
+func (e ErrPropagateMultipleCommitsInRepo) Error() string {
+	return fmt.Sprintf("cannot propgate multiple commits off of repo %q. Previous head: %q. InconsisentHead: %q. Branches: %q. BranchHeads: %q. CommitInfo: %q. ID: %q.", e.Repo, e.PreviousHead, e.InconsisentHead, e.Branches, e.BranchHeads, e.CommitInfo, e.ID)
```

Comment:
Should not be separate sentences, and perhaps should not be so lengthy (let end-user tooling interpret it).

Also: ‘propagate’ and ‘InconsistentHead.’